app/services/analytics_events.rb

```diff
@@ -5182,14 +5212,15 @@ def user_registration_2fa_setup_visit(
   end
 
   # User registration has been handed off to agency page
-  # @param [Boolean] ial2
-  # @param [Integer] ialmax
-  # @param [String] service_provider_name
-  # @param [String] page_occurence
-  # @param [String] needs_completion_screen_reason
+  # @param [Boolean] ial2 Whether the user registration was for a verified identity
+  # @param [Integer] ialmax Whether the user registration was for an IALMax request
+  # @param [String] service_provider_name The friendly name of the service provider
+  # @param ['account-page','agency-page'] page_occurence Where the user concluded registration
+  # @param ['new_sp','new_attributes','reverified_after_consent'] needs_completion_screen_reason The
+  # reason for the consent screen being shown
   # @param [Boolean] in_account_creation_flow Whether user is going through account creation
-  # @param [Array] sp_request_requested_attributes
-  # @param [Array] sp_session_requested_attributes
+  # @param [Array] sp_request_requested_attributes Attributes requested by the service provider
+  # @param [Array] sp_session_requested_attributes Attributes requested by the service provider
```

Comment:
can we describe what the difference is between these two? I'm guessing one is per-request and the other is something else?

Reply:
I looked into this a bit, and... I think they're the same? I'll double-check though. And if they are the same, maybe I'll follow-up or create a ticket to remove one or the other.

Reply:
I'm not seeing anything in the logs for the `_request_` one in the last 2 weeks... very mysterious

```
| filter ispresent(properties.event_properties.sp_request_requested_attributes) or ispresent(properties.event_properties.sp_request_requested_attributes.0)
```

Reply:
I think the issue is [that value reads from `service_provider_request`](https://github.com/18F/identity-idp/blob/66f3025e31b2c3d38b841a78944618351d1fc846/app/controllers/sign_up/completions_controller.rb#L87), but [`service_provider_request` relies on the query parameter](https://github.com/18F/identity-idp/blob/66f3025e31b2c3d38b841a78944618351d1fc846/app/controllers/application_controller.rb#L205), which we wouldn't expect to be present in the completions controller.

I can't imagine that `service_provider_request` has much utility to be honest, and we should probably be calling `sp_session` in most places instead. I can see about following-up on that.